internal/api/kubernetes/client.go

```diff
@@ -553,7 +553,7 @@ func getAuthorizedClient(globalServiceAccountNamespaces []string) func(
 			Name:        key.Name,
 		}
 
-		if userInfo.Username != "" {
+		if userInfo.Subject != "" {
```

Comment:
Do we need to consider Email too?

Reply:
Good question. "sub" and "email" are both standard claims. We'll either have a value for both or for neither. So we only need to check one to know what kind of scenario we're dealing with.